source/server-discovery-and-monitoring/server-discovery-and-monitoring.md

```diff
@@ -921,8 +931,11 @@ checkIfHasPrimary()
 ```
 
 A note on invalidating the old primary: when a new primary is discovered, the client finds the previous primary (there
-should be none or one) and replaces its description with a default ServerDescription of type "Unknown." A multi-threaded
-client MUST [request an immediate check](server-monitoring.md#requesting-an-immediate-check) for that server as soon as
+should be none or one) and replaces its description with a default ServerDescription of type "Unknown". Additionally,
+the `error` field of the new `ServerDescription` object MUST include a descriptive error explaining that it was
+invalidated because the primary was determined to be stale. Drivers MAY additionally specify whether this was due to an
```

Comment:
This paragraph is "a note on invalidating the old primary" so I don't think it applies to all the "electionId/setVersion mismatch" case and this sentence should be moved:
> Drivers MAY additionally specify whether this was due to an electionId or setVersion mismatch as described in the [ServerDescripion.error section](#error).

It should be moved to the paragraph below that starts with "If the server is primary with an obsolete electionId or setVersion,"